Makefile

```diff
@@ -63,15 +71,25 @@ lib/libbase64.o: $(OBJS)
 	$(OBJCOPY) --keep-global-symbols=lib/exports.txt $@
 
 lib/config.h:
-	@echo "#define HAVE_AVX2   $(HAVE_AVX2)"    > $@
-	@echo "#define HAVE_NEON32 $(HAVE_NEON32)" >> $@
-	@echo "#define HAVE_NEON64 $(HAVE_NEON64)" >> $@
-	@echo "#define HAVE_SSSE3  $(HAVE_SSSE3)"  >> $@
-	@echo "#define HAVE_SSE41  $(HAVE_SSE41)"  >> $@
-	@echo "#define HAVE_SSE42  $(HAVE_SSE42)"  >> $@
-	@echo "#define HAVE_AVX    $(HAVE_AVX)"    >> $@
+	@echo "#define HAVE_AVX2                  $(HAVE_AVX2)"                   > $@
+	@echo "#define HAVE_NEON32                $(HAVE_NEON32)"                >> $@
+	@echo "#define HAVE_NEON64                $(HAVE_NEON64)"                >> $@
+	@echo "#define HAVE_SSSE3                 $(HAVE_SSSE3)"                 >> $@
+	@echo "#define HAVE_SSE41                 $(HAVE_SSE41)"                 >> $@
+	@echo "#define HAVE_SSE42                 $(HAVE_SSE42)"                 >> $@
+	@echo "#define HAVE_AVX                   $(HAVE_AVX)"                   >> $@
+	@echo "#define HAVE_FAST_UNALIGNED_ACCESS $(HAVE_FAST_UNALIGNED_ACCESS)" >> $@
+
+lib/table_generator: lib/table_generator.o
+	$(CC) $(CFLAGS) -o $@ $^
+
+lib/tables.h: lib/table_generator.c
+	@$(MAKE) -f $(THIS_FILE) lib/table_generator
+	./lib/table_generator > $@
+
```

Comment:
Hmm, yes you're right of course. My proposed change would force a rebuild after a fresh checkout, which is the thing we wanted to avoid by putting the tables under source control...

The reason why I proposed the change was that I think the `$(THIS_FILE)` construction is an ugly hack. Can we avoid it in any way? The following thing works (does not rebuild needlessly), at the cost of repeating the recipe for `%.o`, but at the gain of getting rid of the rule for `lib/table_generator`:

```make
lib/tables.h: ./lib/table_generator.c
        $(CC) $(CFLAGS) -o lib/table_generator $^
        ./lib/table_generator > $@
```